apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js

```diff
@@ -499,78 +524,246 @@ function saveAnnotations() {
+// 					addAnnotations( data, layer );
+// 				}
+// 				// searchForAnnotations();
+// 			})
+// 			.fail( function() {
+// 				// searchForAnnotations();
+// 			});
+//
+//
+// 			return false;
+// 	});
+// }
+
+function addSavedAnnotations( geoJsonSource, layer ) {
+	var features = new ol.format.GeoJSON().readFeatures( geoJsonSource );
+	
+	$.each( features, function( index, feature ) {
+		var geometry = new ol.format.WKT().readGeometry(
+			feature.getProperties().geometry_ortho, {
+				dataProjection: 'EPSG:4326',
+				featureProjection: 'EPSG:3857'
+			} );
+		
+		feature.setGeometry( geometry );
+	} );
+	
+	createAnnotationsLayer( layer );
+	
+	var source = layer.annotationsLayer.getSource();
+	
+	source.un( 'addfeature', anAnnotationHasBeenAdded );
+	
+	layer.annotationsLayer.getSource().addFeatures( features );
+	
+	source.on( 'addfeature', anAnnotationHasBeenAdded );
+	
+	$.each( source.getFeatures(), function( index, feature ) {
+		var properties = feature.getProperties();
+		var style      = properties.style || createDefaultStyle();
+		var label      = properties.type + '\n' + properties.username;
+		
+		style.getText().setText( label );
+		feature.setStyle( style );
+	} );
+}
 
+function generateRandomNumber( min, max ) {
+	return ~~( Math.random() * max ) + min;
+}
 
-			return false;
-		}
-	});
+function generateRandomColor( min, max, alpha ) {
+	return 'rgba(' +
+		generateRandomNumber( min, max ) + ', ' +
+		generateRandomNumber( min, max ) + ', ' +
+		generateRandomNumber( min, max ) + ', ' +
+		alpha + ')';
+}
+
+function generateRandomStyle( min, max ) {
+	var alpha0 = generateRandomColor( min, max, 0 );
+	var alpha1 = generateRandomColor( min, max, 1 );
+	
+	return new ol.style.Style( {
+		geometry: function( feature ) {
+			var geometry = feature.getGeometry();
+			if( geometry.getType() === 'MultiPolygon' ) {
+				// Only render label for the widest polygon of a multipolygon
+				var polygons = geometry.getPolygons();
+				var widest   = 0;
+				for( var i = 0, ii = polygons.length; i < ii; ++i ) {
+					var polygon = polygons[ i ];
+					var width   = ol.extent.getWidth( polygon.getExtent() );
+					if( width > widest ) {
+						widest   = width;
+						geometry = polygon;
+					}
+				}
+			}
+			
+			
+			return geometry;
+		},
+		fill: new ol.style.Fill( {
+			color: alpha0
+		} ),
+		image: new ol.style.Circle( {
+			fill: new ol.style.Fill( {
+				color: alpha1
+			} ),
+			radius: 5,
+			stroke: new ol.style.Stroke( {
+				color: alpha0,
+				width: 2
+			} )
+		} ),
+		stroke: new ol.style.Stroke( {
+			color: alpha1,
+			width: 2
+		} ),
+		text: new ol.style.Text( {
+			fill: new ol.style.Fill( {
+				color: alpha1
+			} ),
+			offsetY: -13,
+			overflow: true
+		} )
+	} );
+}
+
+function searchForAnnotations() {
+	var usernames = {};
+	
+	tlv.layers.forEach( function( layer ) {
+		console.log( layer );
+		
+		$.ajax( {
+			url: '/annotation/imageAnnotations' + '?' + $.param( { imgId: layer.imageId } )
```

Comment:
It looks like you eventually create a GeoJSON object from the data anyway, why is a custom endpoint needed to get the data when WFS will do all the conversion work for you?

Reply:
I was trying to get the annotations applied to an image and display them. Are you saying I can get that from WFS?

Reply:
Yeah, the WFS will return all the annotations for a specific image in GeoJSON format. The issue, for testing at least, is that the features will be the ones on the server, not the ones you've created locally. But you can always just used the dev instance, create some features, save them and then do all your testing locally.